nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java

```diff
@@ -131,14 +129,29 @@ public void postConstruct() {
     @Override
     public void event(Event event) {
         if (event.is(Deployment.ALL_APPLICATIONS_LOADED)) {
-            // there is awkward point in embedded EJBContainer initialization, where same server
-            // instance is started twice. In such case, we need to reinitialize.
-            if (threadPoolExecutor.isShutdown()) {
-                initialiseThreadPools();
-            }
+            // Embedded containers can be started and stopped multiple times.
+            // Thus we need to initialize anytime the server instance is started.
+            initialiseThreadPools();
         } else if (event.is(EventTypes.SERVER_SHUTDOWN)) {
             threadPoolExecutor.shutdown();
```

Comment:
Maybe have a method like `terminateThreadPools` for this?

Reply:
Seems reasonable ... and done.